src/text2.c

```diff
@@ -3210,3 +3210,860 @@ void sub_80089AC(const UnkTextStruct2 *r4, UnkTextStruct2_sub *r5_Str)
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+            *(ptr++) = 0;
+        }
+        for (i = 0; i < count; i += 32) {
+            CpuClear(ptr, 32);
+            ptr += 8;
+        }
+    }
+
+    if (strPtr->unkC == 6) {
+        strPtr->unk46 = 1;
+        sub_8007958(strArr, strId, 0, 10, strPtr->unk4 * 8, 0xE);
+        sub_8007958(strArr, strId, 0, 11, strPtr->unk4 * 8, 0xD);
+        strPtr->unk46 = 0;
+    }
+
+    strPtr->unk30 = strPtr->unk28;
+    strPtr->unk34 = strPtr->unk1C;
+    strPtr->unk38 = strPtr->unk2C;
+    strPtr->unk44 = 1;
+}
+
+bool8 sub_8008D8C(u32 strId)
+{
+    UnkTextStruct1 *strPtr = &gUnknown_2027370[strId];
+    return (strPtr->unk38 != 0);
+}
+
+UNUSED bool8 sub_8008DA8(void)
+{
+    return FALSE;
+}
+
+UNUSED void sub_8008DAC(s32 a0, s32 a1, s32 a2)
+{
+    gUnknown_202B038[0][a1][a0] = a2;
+    gUnknown_20274A5 = 1;
+}
+
+void sub_8008DC8(s32 a0, s32 a1, s32 a2, s32 a3)
+{
+    gUnknown_202B038[0][a1][a0] = a2;
+    gUnknown_202B038[1][a1][a0] = a3;
+    gUnknown_20274A5 = 1;
+}
+
+UNUSED void sub_8008DF4(s32 a0, s32 a1, u8 *a2)
+{
+    u8 r1;
+
+    while (1) {
+        if (*a2 == 0)
+            break;
+
+        r1 = *a2;
+        if (r1 >= 97 && r1 <= 122) {
+            r1 -= 32;
+        }
+        gUnknown_202B038[0][a1][a0] = 0xF000 | (r1 + 0x258);
+        a2++;
+        a0++;
+    }
+    gUnknown_20274A5 = 1;
+}
+
+UNUSED void sub_8008E58(s32 a0, s32 a1, u8 *a2, s32 a3)
+{
+    u8 r1;
+
+    while (1) {
+        if (*a2 == 0)
+            break;
+
+        r1 = *a2;
+        if (a3 == 0)
+            break;
+
+        if (r1 >= 97 && r1 <= 122) {
+            r1 -= 32;
+        }
+        gUnknown_202B038[0][a1][a0] = 0xF000 | (r1 + 0x258);
+        a2++;
+        a0++;
+        a3--;
+    }
+    gUnknown_20274A5 = 1;
+}
+
+extern s16 gCharacterSpacing;
+
+s32 sub_8008ED0(const u8 *str)
+{
+    s32 ret = 0;
+
+    while (1) {
+        const u8 *strPtr;
+        u32 chr;
+
+        strPtr = xxx_get_next_char_from_string(str, &chr);
+        str = strPtr;
+        if (chr == 0)
+            break;
+        if (chr == 0xD)
+            break;
+        if (chr == 0xA)
+            break;
+        if (chr == 0x82A0 || chr == 0x82A2)
+            continue;
+        if (chr == 0x1B)
+            break;
+
+        if (chr == 0x23) {
+            if (*strPtr == 0x6E || *strPtr == 0x5B || *strPtr == 0x50)
+                break;
+            switch (*strPtr) {
+                case 0x3D:
+                case 0x7E:
+                    str = strPtr + 2;
+                    break;
+                case 0x43:
+                case 0x63:
+                    str += 2;
+                    break;
+                case 0x52:
+                case 0x72:
+                    str++;
+                    break;
+                case 0x2B:
+                case 0x57:
+                    str = strPtr + 1;
+                    break;
+            }
+        }
+        else if (chr == 0x60) {
```

Comment:
Most of these `chr` comparisons in the last three functions should use actual chars, since they are meaningful characters.
You can use `L'x'` to make the char constant have a u32 value.